packages/utils/telemetry-utils/src/genLogger.ts

```diff
@@ -0,0 +1,204 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { v4 as uuid } from "uuid";
+import {
+    ITelemetryGenApiEvent,
+    ITelemetryGenClassEvent,
+    ITelemetryGenErrorEvent,
+    ITelemetryGenServiceEvent,
+    ITelemetryGenEvent,
+    ITelemetryLogger,
+    TelemetryErrorCategory,
+} from "@fluidframework/common-definitions";
+
+
+export interface IClassEventConfig {
+    docId?: string;
+    clientId?: string;
+    details?: any;
+}
+
+export interface IApiEventConfig {
+    apiName: string;
+    docId?: string;
+    clientId?: string;
+    details?: any;
+}
+
+export interface IErrorEventConfig {
+    errorCode: string;
+    docId?: string;
+    clientId?: string;
+    detastackTraceils?: any;
+    details?: any;
+    severityLevel: TelemetryErrorCategory;
+    message?: string;
+}
+
+export interface IServiceEventConfig {
+    target: string;
+    docId?: string;
+    clientId?: string;
+    details?: any;
+}
+
+/**
+ * Helper class for general-use logs
+ */
+export class GeneralUseLogger {
```

Comment:
Will look at reworking this and exposing event-like class, rather then another "logger" the consumer needs to understand.